### PR TITLE
Add JavaDoc and unit tests to LogEntry class

### PR DESCRIPTION
**Pull request by kumy**
While working on #4911, I've found that LogEntry has no unit tests.
I wrote JavaDoc and some unit tests...

### CODE REVIEW

File: main/src/cgeo/geocaching/LogEntry.java [hunk 1]
```diff
@@ -34,10 +34,26 @@
     public String cacheName = ""; // used for trackables
     public String cacheGuid = ""; // used for trackables
 
+    /**
+     * Construct a LogEntry Object usable for logging Geocaches.
+     * Username field is automatically taken from the current configured user on geocaching.com connector.
+     *
+     * @param dateInMilliSeconds of log
+     * @param type of log
+     * @param text mesage of log
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/LogEntry.java]: Typo ("s" in "message" missing)

File: main/src/cgeo/geocaching/LogEntry.java [hunk 2]
```diff
@@ -45,11 +61,21 @@ public LogEntry(final String author, final long dateInMilliSeconds, final LogTyp
...
+     */
     @Override
     public int hashCode() {
         return (int) date * type.hashCode() * author.hashCode() * log.hashCode();
     }
 
+    /**
+     * Return True if passed LogDate Object is equal to the current LogDate.
+     *
+     * @return the object's hash code
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/LogEntry.java]: Wrong.

File: main/src/cgeo/geocaching/LogEntry.java [hunk 3]
```diff
@@ -110,6 +154,11 @@ public String getDisplayText() {
         return log;
     }
 
+    /**
+     * Check if the LogEntry is owned by the current configured user on geocaching.com connector.
+     *
+     * @return True if LogEntry is from user
+     */
```
Comment (samueltardieu) [inline main/src/cgeo/geocaching/LogEntry.java]: Independently of the connector? Ouch, this is so wrong.

Comment (kumy) [inline main/src/cgeo/geocaching/LogEntry.java]: Do you mean my analysis of `R.string.pref_username` is wrong? :/

**Pull request merged**